subsys/ipc/ipc_service/backends/ipc_icbmsg.c

```diff
@@ -1117,10 +1121,15 @@ static int backend_init(const struct device *instance)
 {
 	const struct icbmsg_config *conf = instance->config;
 	struct backend_data *dev_data = instance->data;
+	static K_THREAD_STACK_DEFINE(ep_bound_work_q_stack, EP_BOUND_WORK_Q_STACK_SIZE);
 
 	dev_data->conf = conf;
 	dev_data->is_initiator = (conf->rx.blocks_ptr < conf->tx.blocks_ptr);
 	k_mutex_init(&dev_data->mutex);
+	k_work_queue_init(&dev_data->ep_bound_work_q);
```

Comment:
Maybe we could keep the workq common among devices? (then we should keep it global and initialize once) Alternatively we would need to define separate stacks for the workqueues (current implementation may lead to really tricky errors I think)

Reply:
With the common workq, we could also easily add a temporary Kconfig to avoid increasing memory usage if not needed (app that needs the workaround would explicitly enable it)

Reply:
You could add an experimental Kconfig to avoid deprecation process I think (anyway I don't think it's necessary)